bundles/org.openhab.binding.mielecloud/src/main/java/org/openhab/binding/mielecloud/internal/config/servlet/CreateBridgeServlet.java

```diff
@@ -0,0 +1,194 @@
+                    + SuccessServlet.BRIDGE_UID_PARAMETER_NAME + "=" + bridgeUidString;
+        }
+    }
+
+    private Thing pairOrReconfigureBridge(String accessToken, String locale, ThingUID bridgeUid) {
+        DiscoveryResult result = DiscoveryResultBuilder.create(bridgeUid)
+                .withRepresentationProperty(Thing.PROPERTY_MODEL_ID).withLabel(MIELE_CLOUD_BRIDGE_LABEL)
+                .withProperty(Thing.PROPERTY_MODEL_ID, MIELE_CLOUD_BRIDGE_NAME).withProperty("accessToken", accessToken)
+                .withProperty(LOCALE_PARAMETER_NAME, locale).build();
+        if (inbox.add(result)) {
+            return pairBridge(bridgeUid);
+        } else {
+            return reconfigureBridge(bridgeUid, locale);
+        }
+    }
+
+    private Thing pairBridge(ThingUID thingUid) {
+        Thing thing = inbox.approve(thingUid, MIELE_CLOUD_BRIDGE_LABEL, null);
+        if (thing == null) {
+            throw new BridgeCreationFailedException();
+        }
+
+        logger.info("Successfully created bridge {}", thingUid);
+        return thing;
+    }
+
+    private Thing reconfigureBridge(ThingUID thingUid, String locale) {
+        logger.info("Thing already exists. Modifying configuration.");
+        Thing thing = thingRegistry.get(thingUid);
+        if (thing == null) {
+            throw new BridgeReconfigurationFailedException(
+                    "Cannot modify non existing bridge: Could neither add bridge via inbox nor find existing bridge.");
+        }
+
+        ThingHandler handler = thing.getHandler();
+        if (handler == null) {
+            throw new BridgeReconfigurationFailedException("Bridge exists but has no handler.");
+        }
+
+        handler.handleConfigurationUpdate(
+                Collections.singletonMap(MieleCloudBindingConstants.CONFIG_PARAM_LOCALE, locale));
+
+        // As the parameters will not necessarily change we need to force the thing to re-initialize.
+        handler.dispose();
+        handler.initialize();
+
+        return thing;
+    }
+
+    private String getValidLocale(@Nullable String localeParameterValue) {
+        if (localeParameterValue == null || localeParameterValue.isEmpty()
+                || !LocaleValidator.isValidLanguage(localeParameterValue)) {
+            return DEFAULT_LOCALE;
+        } else {
+            return localeParameterValue;
+        }
+    }
+
+    private void waitForBridgeToComeOnline(Thing bridge) {
+        long remainingWaitTime = ONLINE_WAIT_TIMEOUT_IN_MILLISECONDS;
+        while (bridge.getStatus() != ThingStatus.ONLINE && remainingWaitTime > 0) {
+            try {
+                TimeUnit.MILLISECONDS.sleep(ONLINE_CHECK_INTERVAL_IN_MILLISECONDS);
+                remainingWaitTime -= ONLINE_CHECK_INTERVAL_IN_MILLISECONDS;
+            } catch (InterruptedException e) {
+                return;
+            }
+        }
+
+        try {
+            TimeUnit.SECONDS.sleep(1);
+        } catch (InterruptedException e) {
+        }
```

Comment:
The delay is 1s which is larger than 100ms :wink: 

You are right, the purpose is not that obvious here: After our bridge changed to `ONLINE` (see the sleep above) it might take a moment for our discovery service to initialize and discover all devices. We wait for this because the overview page which the user is redirected to shows a template for a `.things` file which is populated with the discovery results. If these are missing then the template is incomplete. We want to avoid confusing the user here. Obviously that is a hacky workaround and we should rather wait until a maximum delay expired or the number of discovered things in the inbox doesn't change any more. Or is there a more common / easier solution?

Reply:
It seems it was a bit too late that day...

It'd be good if you solve this in a more elegant way. Unfortunately I can't come up with a solution.

Reply:
We now wait until the number of discovery results from this binding doesn't change any more.